### DISCUSSION TITLE
V400 goes out of range with provided start/end gcodes or prints in mid air if default code is used

### DISCUSSION THREAD

**Discussion by PunCala** [Guilouz/Klipper-Flsun-Speeder-Pad#113] [Q&A]
![Sieppaa](https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/assets/135166060/d1fbadd6-0e5a-4e3f-9f01-3912914ef681)

I get this error if I try to print anything using the start and end gcodes that were in the manual. It says that the printer is out of range. If I try to use the default V400 codes that come with Cura and OrcaSlicer, the device prints in mid air, like 3 cm above the bed. Furthermore, in the latter case the temperature is somehow changed from 260C to 150C. Can anyone help me figure out what to do?

**Comment by PunCala** [comment]
This is the macro that this installation process makes (I commented it for myself as I am a beginner with regards to gcode). It is completely wrong:

[gcode_macro START_PRINT]
description: Start G-Code
gcode:
{% set BED_TEMP = params.BED_TEMP|default(60)|float %} #set bed temp, same as M140
{% set EXTRUDER_TEMP = params.EXTRUDER_TEMP|default(210)|float %} #set extruder temp, same as M104
LED_HOTEND_ON #turn on hotend LED
G21 #set units to mm G90 #absolute coordinates (G91 = relative)
M82 #absolute extrude mode (or M83?)
{% if printer.extruder.target != 0 %} #beginning of if statement
M104 S{EXTRUDER_TEMP} #set extruder temp
{% else %} #else statement
M104 S150 #set extruder temp, WHY 150C?
{% endif %} #end if
M140 S{BED_TEMP} #set bed temp
{% if printer.toolhead.homed_axes != "xyz" %} #???
G28 #move to origin
{% endif %} #end if statement
G1 Z370 F3000 #move to position Z 370 mm at speed 3000
G1 X-150 F3000 #move to position X -150 mm at speed 3000 M190 S{BED_TEMP} #set bed temp and wait
M109 S{EXTRUDER_TEMP} #set extruder temp and wait to settle
G1 Z1 F3000 #move to position Z 1 mm at speed 3000
G1 X-150 Y0 Z0.4 #move to X-150, Y0, Z0.4
G92 E0 #set position
G3 X0 Y-130 I150 J1 Z0.3 E30 F2000 #arc move command, requires [gcode_arcs] in config
G92 E0 #set position x y z e, where e is extruded length
G1 E-1 F2100 #move to position, retract 1 mm?

**Comment by lukegoldmeyer** [comment]
I'm having the same issue here....
